src/launch/main.c

```diff
@@ -801,6 +808,11 @@ static int manager_on_set_activation_environment(Manager *manager, sd_bus_messag
         _c_cleanup_(sd_bus_message_unrefp) sd_bus_message *method_call = NULL;
         int r;
 
+        if (main_arg_scope == MANAGER_SCOPE_NONE) {
+                fprintf(stderr, "Ignoring request to set activation environment outside of scope.\n");
+                return 0;
```

Comment:
Errr, this should not print a message. We should just pretend to honor the call and silently return.